### ISSUE TITLE
Ops added after calling start_queue_runners() are broken

### ISSUE THREAD

**Issue by MarkDaoust** [tensorflow/tensorflow#945]
(Using the OSX CPU-only  v.0.6 wheel). 

I think this is a minimal example:

```
image = tf.constant(0)
images = tf.train.shuffle_batch([image], batch_size=1, capacity=100, min_after_dequeue=1)

a = tf.constant(1)
with tf.Session():
    b = a+1
    tf.train.start_queue_runners()
    c = a+1

    print(a.eval())
    print(b.eval())
    print(c.eval())
```

It *usually* crashes on the `c.eval()` giving the following error:
```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    print(c.eval())
  File ".../python2.7/site-packages/tensorflow/python/framework/ops.py", line 460, in eval
    return _eval_using_default_session(self, feed_dict, self.graph, session)
  File ".../python2.7/site-packages/tensorflow/python/framework/ops.py", line 2910, in _eval_using_default_session
    return session.run(tensors, feed_dict)
  File ".../python2.7/site-packages/tensorflow/python/client/session.py", line 368, in run
    results = self._do_run(target_list, unique_fetch_targets, feed_dict_string)
  File ".../python2.7/site-packages/tensorflow/python/client/session.py", line 446, in _do_run
    six.reraise(e_type, e_value, e_traceback)
  File ".../python2.7/site-packages/tensorflow/python/client/session.py", line 428, in _do_run
    target_list)
tensorflow.python.pywrap_tensorflow.StatusNotOK: Not found: FetchOutputs node add_1:0: not found
```
Even just a clearer error message in this situation would be a big help.

**Comment by josh11b** [comment]
The issue is that start_queue_runners() starts (one or more) threads that are running portions of the graph.  Since graph modification is not thread-safe, you will see problems if the graph is modified after start_queue_runners().  @mrry do we have a way of making this an error?

**Comment by mrry** [comment]
We could use [`tf.Graph.finalize()`](https://www.tensorflow.org/versions/0.6.0/api_docs/python/framework.html#Graph.finalize), which is intended for such purposes. To the original point: the `tf.Graph` methods for adding ops are definitely not thread-safe, and it would require a big redesign to achieve that (with only a small upside).

**Comment by MarkDaoust** [comment]
An error would be nice, it could save other beginners a headache.

Thanks for the clarifications.

**Comment by martinwicke** [comment]
Sounds like `start_queue_runners` should call `finalize` (possibly with a flag to turn it off if you really know what you're doing)?

**Comment by girving** [comment]
@josh11b: Should I reassign?

**Comment by mrry** [comment]
I think this is fixed since acac487ac4ebaa6edb3e3f866d41cbd12546a107.
